compiler-core/src/language_server/configuration.rs

```diff
@@ -0,0 +1,25 @@
+use serde::Deserialize;
+use std::sync::{Arc, RwLock};
+
+pub type SharedConfig = Arc<RwLock<Configuration>>;
+
+#[derive(Debug, Default, Clone, Deserialize, PartialEq, Eq)]
+#[serde(default)]
+#[serde(rename_all = "camelCase")]
+pub struct Configuration {
```

Comment:
This is called configuration here but "user config" elsewhere. Is there a canonical name for this sort of configuration in the protocol?